ubl/Settings/TaxesSettings.php

```diff
@@ -96,27 +99,53 @@ public function output_table_for_tax_class( $slug, $name ) {
 				<tr>
 					<th colspan="5" style="text-align: right;"><?php _e( 'Tax class default', 'woocommerce-pdf-invoices-packing-slips' ); ?>:</th>
 					<?php
-						$scheme   = isset( $this->settings['class'][ $slug ]['scheme'] ) ? $this->settings['class'][ $slug ]['scheme'] : '';
+						$scheme   = isset( $this->settings['class'][ $slug ]['scheme'] )   ? $this->settings['class'][ $slug ]['scheme']   : '';
 						$category = isset( $this->settings['class'][ $slug ]['category'] ) ? $this->settings['class'][ $slug ]['category'] : '';
+						$reason   = isset( $this->settings['class'][ $slug ]['reason'] )   ? $this->settings['class'][ $slug ]['reason']   : '';
 					?>
-					<th><?php echo $this->get_scheme_select( 'class', $slug, $scheme ); ?></th>
-					<th><?php echo $this->get_category_select( 'class', $slug, $category ); ?></th>
+					<th><?php echo $this->get_select_for( 'scheme', 'class', $slug, $scheme ); ?></th>
+					<th><?php echo $this->get_select_for( 'category', 'class', $slug, $category ); ?></th>
+					<th><?php echo $this->get_select_for( 'reason', 'class', $slug, $reason ); ?></th>
 				</tr>
 			</tfoot>
 		</table>
 		<?php
 	}
 
-	public function get_scheme_select( $type, $id, $selected ) {
-		$select = '<select name="wpo_wcpdf_settings_ubl_taxes['.$type.']['.$id.'][scheme]"><option value="">' . __( 'Default', 'woocommerce-pdf-invoices-packing-slips' ) . '</option>';
-		foreach ( $this->get_available_schemes() as $key => $value ) {
-			$select .= '<option '.selected( $key, $selected, false ).' value="'.$key.'">'.$value.'</option>';
+	/**
+	 * Get select field for tax rate
+	 *
+	 * @param string $for
+	 * @param string $type
+	 * @param string $id
+	 * @param string $selected
+	 *
+	 * @return string
+	 */
+	public function get_select_for( string $for, string $type, string $id, string $selected ): string {
+		switch ( $for ) {
+			case 'scheme':
+				$options = $this->get_available_schemes();
+				break;
+			case 'category':
+				$options = $this->get_available_categories();
+				break;
+			case 'reason':
+				$options = self::get_available_reasons();
+				break;
+			default:
+				$options = array();
+		}
+
+		$select = '<select name="wpo_wcpdf_settings_ubl_taxes[' . $type . '][' . $id . '][reason]"><option value="">' . __( 'Default', 'woocommerce-pdf-invoices-packing-slips' ) . '</option>';
```

Comment:
The `reason` should be replaced with `$for`:
`wpo_wcpdf_settings_ubl_taxes[' . $type . '][' . $id . '][reason]`

Reply:
Good catch!